pandas-stubs/core/frame.pyi

```diff
@@ -558,7 +559,7 @@ class DataFrame(NDFrame, OpsMixin):
         limit: int | None = ...,
         fill_axis: Axis = ...,
         broadcast_axis: Axis | None = ...,
-    ) -> DataFrame: ...
+    ) -> tuple[DataFrame, _SeriesOrFrameT]: ...
```

Comment:
Agree to use `NDFrameT`, because that will give the precise type of `Series` or `DataFrame` based on `other`

Tests should test using `other` as a `DataFrame` and `other` as a `Series`